lib/ansible/modules/windows/win_dhcp_lease.ps1

```diff
@@ -0,0 +1,473 @@
+    options = @{
+        type = @{ type = "str"; choices = "reservation", "lease"; default = "reservation" }
+        ip = @{ type = "str" }
+        scope_id = @{ type = "str" }
+        mac = @{ type = "str" }
+        duration = @{ type = "int" }
+        dns_hostname = @{ type = "str"; }
+        dns_regtype = @{ type = "str"; choices = "aptr", "a", "noreg"; default = "aptr" }
+        reservation_name = @{ type = "str"; }
+        description = @{ type = "str"; }
+        state = @{ type = "str"; choices = "absent", "present"; default = "present" }
+    }
+    supports_check_mode = $true
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+$module.Result.changed = $false
+$check_mode = $module.CheckMode
+
+$type = $module.Params.type
+$ip = $module.Params.ip
+$scope_id = $module.Params.scope_id
+$mac = $module.Params.mac
+$duration = $module.Params.duration
+$dns_hostname = $module.Params.dns_hostname
+$dns_regtype = $module.Params.dns_regtype
+$reservation_name = $module.Params.reservation_name
+$description = $module.Params.description
+$state = $module.Params.state
+
+Function Convert-MacAddress {
+    Param(
+        [string]$mac
+    )
+
+    # Evaluate Length
+    if ($mac.Length -eq 12) {
+        # Insert Dashes
+        $mac = $mac.Insert(2, "-").Insert(5, "-").Insert(8, "-").Insert(11, "-").Insert(14, "-")
+        return $mac
+    }
+    elseif ($mac.Length -eq 17) {
+        # Remove Dashes
+        $mac = $mac -replace '-'
+        return $mac
+    }
+    else {
+        return $false
+    }
+}
+
+Function Compare-DhcpLease {
+    Param(
+        [PSObject]$Original,
+        [PSObject]$Updated
+    )
+
+    # Compare values that we care about
+    if (($Original.AddressState -eq $Updated.AddressState) -and ($Original.IPAddress -eq $Updated.IPAddress) -and ($Original.ScopeId -eq $Updated.ScopeId) -and ($Original.Name -eq $Updated.Name) -and ($Original.Description -eq $Updated.Description)) {
+        # changed = false
+        return $false
+    }
+    else {
+        # changed = true
+        return $true
+    }
+}
+
+Function Convert-ReturnValue {
+    Param(
+        $Object
+    )
+
+    $data = @{
+        address_state = $Object.AddressState
+        client_id     = $Object.ClientId
+        ip_address    = $Object.IPAddress.IPAddressToString
+        scope_id      = $Object.ScopeId.IPAddressToString
+        name          = $Object.Name
+        description   = $Object.Description
+    }
+
+    return $data
+}
+
+# Parse Regtype
+if ($dns_regtype) {
+    Switch ($dns_regtype) {
+        "aptr" { $dns_regtype = "AandPTR"; break }
+        "a" { $dns_regtype = "A"; break }
+        "noreg" { $dns_regtype = "NoRegistration"; break }
+        default { $dns_regtype = "NoRegistration"; break }
+    }
+}
+
+Try {
+    # Import DHCP Server PS Module
+    Import-Module DhcpServer
+}
+Catch {
+    # Couldn't load the DhcpServer Module
+    $module.FailJson("The DhcpServer module failed to load properly")
```

Comment:
You can add a 2nd argument to `FailJson()` that is either the error record or exception and the output returned back to Ansible will include the error info

```suggestion
    $module.FailJson("The DhcpServer module failed to load properly", $_)
```